graphql/src/main/java/com/scalar/db/graphql/datafetcher/QueryScanDataFetcher.java

```diff
@@ -42,7 +44,15 @@ public QueryScanDataFetcher(DistributedStorage storage, DataFetcherHelper helper
     LOGGER.debug("got scan argument: {}", scanInput);
     Scan scan = createScan(scanInput);
 
-    // TODO: scan.withProjections()
+    // Add the specified field names, filtered by "*/*", to the scan object as the projections
+    // since the fields of the scan input are represented in the following format:
+    // <object_type_name>_ScanPayLoad.<object_type_name>/<object_type_name>.<field_name>
+    List<String> selectedFieldNames =
```

Comment:
Do we need to do the same thing for Get operations?

Reply:
I did not notice that `Get` operations also had projections. Updated to do the same for them as `Scan`s, and did some refactoring. Please take a look again. Thank you!